client/cloudSubnet_service.go

```diff
@@ -64,33 +82,54 @@ func (sm *ServiceManager) DeleteCloudSubnet(ip string, cloud_cidr_pool_addr stri
 	return sm.DeleteByDn(dn, models.CloudsubnetClassName)
 }
 
-func (sm *ServiceManager) UpdateCloudSubnet(ip string, cloud_cidr_pool_dn string, description string, cloudSubnetattr models.CloudSubnetAttributes, zoneDn string) (*models.CloudSubnet, error) {
+func (sm *ServiceManager) UpdateCloudSubnet(ip string, cloud_cidr_pool_dn string, description string, cloudSubnetattr models.CloudSubnetAttributes, zoneDn, cloudRsSubnetToCtx string) (*models.CloudSubnet, error) {
 	rn := fmt.Sprintf("subnet-[%s]", ip)
-	// parentDn := fmt.Sprintf("uni/tn-%s/ctxprofile-%s/cidr-[%s]", tenant, cloud_context_profile, cloud_cidr_pool_addr)
 	parentDn := cloud_cidr_pool_dn
 	cloudSubnet := models.NewCloudSubnet(rn, parentDn, description, cloudSubnetattr)
 	cloudSubnet.Status = "modified"
 
 	jsonPayload, _, err := sm.PrepareModel(cloudSubnet)
+	jsonPayload.Array(cloudSubnet.ClassName, "children")
 
-	rsZoneAttachJSON := []byte(fmt.Sprintf(`
-	{
-		"cloudRsZoneAttach": {
-			"attributes": {
-				"annotation": "orchestrator:terraform",
-				"dn": "%s/%s/rszoneAttach",
-				"tDn": "%s"
+	if zoneDn != "" {
+		rsZoneAttachJSON := []byte(fmt.Sprintf(`
+		{
+			"cloudRsZoneAttach": {
+				"attributes": {
+					"annotation": "orchestrator:terraform",
```

Comment:
don't hardcode it. Do you have access to the parent orchestrator value?

Reply:
We don't have a definition of the orchestrator in the go client flies.
"orchestrator:terraform" is used as the annotation in the definition of all the relationship attributes payload.

Reply:
Opened https://github.com/ciscoecosystem/aci-go-client/issues/266 to track this in the future